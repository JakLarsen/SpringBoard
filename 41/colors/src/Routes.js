import {Route, Redirect} from 'react-router-dom'

import Color from './Color'
import ColorForm from './ColorForm'
import Home from './Home'




const Routes = () => {

    return (
        <>
            {/* Home isn't rendering?? when we have the Redirect?? */}
            <Route exact path='/'>
                <Home/>
            </Route>
            <Route exact path='/colors'>
                <ColorForm/>
            </Route>
            <Route exact path='/colors/:color'>
                <ColorForm/>
                <Color/>
            </Route>
            <Redirect to='/colors'/>
        </>
    )
}

export default Routes