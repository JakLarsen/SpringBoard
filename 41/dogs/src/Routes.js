import {BrowserRouter, Route, Switch, Redirect} from 'react-router-dom'

import Dogs from './Dogs';
import Dog from './Dog';



const Routes = () => {
     return (
         <>
            <Route exact path='/dogs'><Dogs/></Route>
            <Route exact path='/dogs/:name'><Dog/></Route>
            <Redirect to='/dogs'/>
        </>
     )
}

export default Routes