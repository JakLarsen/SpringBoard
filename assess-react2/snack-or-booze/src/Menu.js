import React from "react";
import { Link } from "react-router-dom";
import "./css/Menu.css";
import MenuItemForm from "./MenuItemForm";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";


/**
 * Menu display for either SnackItem components or DrinkItem components
 * We grab itemType as a toggle for whether to make a Snack menu or Drink menu
 */
function Menu({ items , title, addSnack, addDrink }) {

  let itemType = title.toLowerCase()

  let itemDesc = itemType == 'snacks' ? 
  'Apps and entres of the highest calibre.' : 'Best drinks this side of the river.'

  return (
    <div>
      
        <section>
          <Card>
            <CardBody>
              <CardTitle className="font-weight-bold text-center">
                {`${title} Menu`}
              </CardTitle>
              <CardText>
                {itemDesc}
              </CardText>

              {/* Display each item with a link to their MenuItem component display view */}
              <ListGroup>
                {items.map(item => (
                  <Link to={`/${itemType}/${item.id}`} key={item.id}>
                    <ListGroupItem>{item.name}</ListGroupItem>
                  </Link>
                ))}
              </ListGroup>

            </CardBody>
          </Card>
        </section>

        <MenuItemForm title={title} addSnack={addSnack} addDrink={addDrink}/>
        
    </div>
    


  );
}

export default Menu;
