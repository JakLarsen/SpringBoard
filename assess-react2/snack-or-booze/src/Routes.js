import React, { useState, useEffect } from "react";
import "./css/Routes.css";
import Home from "./Home";
import SnackOrBoozeApi from "./Api";
import { Route, Switch} from "react-router-dom";
import Menu from "./Menu";
import MenuItem from "./MenuItem";
import NotFound from "./NotFound";
import Order from "./Order";
import OrderMenu from "./OrderMenu";

/**
 * Our main Routes, State Management, and Handlers
 */
function Routes() {

 /**
  * State Management
  * 
  * -Loading
  * -snacks
  * -drinks
  * -order
  */
  const [isLoading, setIsLoading] = useState(true);
  const [snacks, setSnacks] = useState([]);
  const [drinks, setDrinks] = useState([]);
  let [order, setOrder] = useState([])


  /**
   * useEffect for retrieving snacks, drinks, and order on initial load
   */
  useEffect(() => {
    async function getSnacks() {
      let snacks = await SnackOrBoozeApi.getSnacks();
      setSnacks(snacks);
      setIsLoading(false);
    }
    async function getDrinks() {
      let drinks = await SnackOrBoozeApi.getDrinks();
      setDrinks(drinks);
      setIsLoading(false);
    }
    async function getOrder(){
      let order = await SnackOrBoozeApi.getOrder();
      setOrder(order)
      setIsLoading(false)
    }
    getSnacks();
    getDrinks();
    getOrder();
  }, []);

  const addSnack = (snack) => {
      setSnacks(snacks => [...snacks, snack ])
      //Could post snack as well
  }
  const addDrink = (drink) => {
      setDrinks(drinks => [...drinks, drink ])
      //Could post drink as well
  }
  const addToOrder = (item) => {
    setOrder(items => [...items, item])
      //Order I thnk I'd keep solely as state while on the site for now
  }

  /**
   * Remove the first item with a given id from your order
   */
  const removeFromOrder = (id) => {

    let yetToRemove = true
    let newOrder = order.map(x=>x)

    newOrder.map((item, idx)=> {
      if(item.id === id && yetToRemove){
        newOrder.splice(idx, 1) 
        yetToRemove = false
      } 
    })
    setOrder(newOrder)
  }

  if (isLoading) {
    return <p>Loading &hellip;</p>;
  }

  return (
    <div className="Routes">
      <Switch>
        <Route exact path="/">
            <Home />
        </Route>
        <Route exact path="/snacks">
            <Menu items={snacks} title="Snacks" addSnack={addSnack}/>
        </Route>
        <Route path="/snacks/:id">
            <MenuItem items={snacks} cantFind="/snacks" />
        </Route>
        <Route exact path="/drinks">
            <Menu items={drinks} title="Drinks" addDrink={addDrink}/>
        </Route>
        <Route path="/drinks/:id">
            <MenuItem items={drinks} cantFind="/drinks" />
        </Route>
        <Route path="/order">
            <Order items={order} title="Order" removeFromOrder={removeFromOrder}/>
            <OrderMenu items={snacks} title="Snacks" addToOrder={addToOrder} />
            <OrderMenu items={drinks} title="Drinks" addToOrder={addToOrder} />
        </Route>
        <Route component={NotFound}/>
      </Switch>
    </div>
  );
}

export default Routes;
