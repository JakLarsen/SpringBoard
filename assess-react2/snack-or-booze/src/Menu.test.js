import React from 'react';
import { render} from '@testing-library/react';
import Menu from './Menu'
import {addSnack, addDrink} from './Routes'
import { BrowserRouter, Route } from "react-router-dom";
import ourTestData from './testData'



let ourRenderView = (
    <BrowserRouter>
            <Route exact path="/snacks">
                <Menu items={ourTestData.snacks} title="Snacks" addSnack={addSnack}/>
            </Route>
    </BrowserRouter>
)

test("renders without crashing", ()=>{
    render(ourRenderView)
})

it('Matches Snack Menu snapshot', function(){
    const {asFragment} = render(
        <BrowserRouter>
            <Route exact path="/snacks">
                <Menu items={ourTestData.snacks} title="Snacks" addSnack={addSnack}/>
            </Route>
        </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
});

// it('Matches Drink Menu snapshot', function(){
//     const {asFragment} = render(
//         <BrowserRouter>
//             <Route exact path="/snacks">
//                 <Menu items={ourTestData.drinks} title="Drinks" addDrinks={addDrink}/>
//             </Route>
//         </BrowserRouter>
//     )
//     expect(asFragment()).toMatchSnapshot()
// });