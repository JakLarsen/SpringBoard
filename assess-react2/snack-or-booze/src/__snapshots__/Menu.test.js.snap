// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snack Menu snapshot 1`] = `<DocumentFragment />`;
